### ISSUE TITLE
hrls files are not trimmed automatically

### ISSUE THREAD

**Issue by cookie33** [EUDAT-B2HANDLE/B2HANDLE-HRLS#46]

**Comment by cookie33** [comment]
We started the reverse lookup servlet on the 31th of August.

Now we have:
```
$ ls logs/hrls-requests.log*
logs/hrls-requests.log           logs/hrls-requests.log-20180824  logs/hrls-requests.log-20180921  logs/hrls-requests.log-20181006
logs/hrls-requests.log-20180731  logs/hrls-requests.log-20180827  logs/hrls-requests.log-20180922  logs/hrls-requests.log-20181007
logs/hrls-requests.log-20180801  logs/hrls-requests.log-20180828  logs/hrls-requests.log-20180923  logs/hrls-requests.log-20181008
logs/hrls-requests.log-20180802  logs/hrls-requests.log-20180829  logs/hrls-requests.log-20180924  logs/hrls-requests.log-20181009
logs/hrls-requests.log-20180803  logs/hrls-requests.log-20180831  logs/hrls-requests.log-20180925  logs/hrls-requests.log-20181010
logs/hrls-requests.log-20180806  logs/hrls-requests.log-20180903  logs/hrls-requests.log-20180926  logs/hrls-requests.log-20181011
logs/hrls-requests.log-20180807  logs/hrls-requests.log-20180904  logs/hrls-requests.log-20180927  logs/hrls-requests.log-20181012
logs/hrls-requests.log-20180810  logs/hrls-requests.log-20180906  logs/hrls-requests.log-20180928  logs/hrls-requests.log-20181013
logs/hrls-requests.log-20180813  logs/hrls-requests.log-20180911  logs/hrls-requests.log-20180929  logs/hrls-requests.log-20181014
logs/hrls-requests.log-20180814  logs/hrls-requests.log-20180912  logs/hrls-requests.log-20180930  logs/hrls-requests.log-20181015
logs/hrls-requests.log-20180815  logs/hrls-requests.log-20180914  logs/hrls-requests.log-20181001  logs/hrls-requests.log-20181016
logs/hrls-requests.log-20180816  logs/hrls-requests.log-20180916  logs/hrls-requests.log-20181002  logs/hrls-requests.log-20181017
logs/hrls-requests.log-20180817  logs/hrls-requests.log-20180917  logs/hrls-requests.log-20181003
logs/hrls-requests.log-20180821  logs/hrls-requests.log-20180919  logs/hrls-requests.log-20181004
logs/hrls-requests.log-20180823  logs/hrls-requests.log-20180920  logs/hrls-requests.log-20181005
```
And it should have automatically trimmed the oldest ones if I understand it correctly.

Is that true or should we trim it ourselves?

We used the the default settings from the hrls servlet setup.
```
$ cat main/resources/log4j2.xml
<?xml version="1.0" encoding="UTF-8"?>
<Configuration status="INFO">
        <Appenders>
                <Console name="Console" target="SYSTEM_OUT">
                        <PatternLayout pattern="%d{DEFAULT} [%c] %-5level %logger{36} - %msg%n" />
                </Console>
                <RollingFile name="requestsLogger" fileName="${env:HANDLE_SVR}/logs/hrls-requests.log" filePattern="${env:HANDLE_SVR}/logs/hrls-requests.log-%d{yyyyMMdd}" createOnDemand="true">
                        <PatternLayout pattern="%d{DEFAULT} [%c] - %msg%n" />
                        <Policies>
                                <TimeBasedTriggeringPolicy />
                        </Policies>
                        <DefaultRolloverStrategy max="28" />
                </RollingFile>
        </Appenders>
        <Loggers>
                <Root level="info">
                        <AppenderRef ref="Console" />
                </Root>
                <Logger name="requestsLogger" level="info" additivity="false">
                        <AppenderRef ref="requestsLogger"/>
                </Logger>
        </Loggers>
</Configuration>
```
We have the daily files. But they just are not trimmed.

**Comment by cookie33** [comment]
This seems to be a feature: https://issues.apache.org/jira/browse/LOG4J2-524

For examples see: http://logging.apache.org/log4j/2.x/manual/appenders.html#CustomDeleteOnRollover


It might need some extra configuring in the log4j2.xml
```
      <DefaultRolloverStrategy>
        <Delete basePath="${env:HANDLE_SVR}/logs/" maxDepth="1">
          <IfFileName glob="hrls-requests.log*" />
          <IfLastModified age="28d" />
        </Delete>
      </DefaultRolloverStrategy>
```

**Comment by cookie33** [comment]
Tested the above. This also does not trim.

**Comment by cookie33** [comment]
At the moment we can trim this using a cronjob. So it is not a deal braker.

**Comment by TobiasWeigel** [comment]
fixed in #49
